src/components/DateRangePicker.jsx

```diff
@@ -283,23 +283,25 @@ export default class DateRangePicker extends React.Component {
   }
 
   responsivizePickerPosition() {
-    const { anchorDirection, horizontalMargin } = this.props;
+    const { anchorDirection, horizontalMargin, withPortal, withFullScreenPortal } = this.props;
     const { dayPickerContainerStyles } = this.state;
 
     const isAnchoredLeft = anchorDirection === ANCHOR_LEFT;
-
-    const containerRect = this.dayPickerContainer.getBoundingClientRect();
-    const currentOffset = dayPickerContainerStyles[anchorDirection] || 0;
-    const containerEdge = isAnchoredLeft ? containerRect[ANCHOR_RIGHT] : containerRect[ANCHOR_LEFT];
-
-    this.setState({
-      dayPickerContainerStyles: getResponsiveContainerStyles(
-        anchorDirection,
-        currentOffset,
-        containerEdge,
-        horizontalMargin
-      ),
-    });
+    if (!withPortal && !withFullScreenPortal) {
+      const containerRect = this.dayPickerContainer.getBoundingClientRect();
+      const currentOffset = dayPickerContainerStyles[anchorDirection] || 0;
+      const containerEdge =
+        isAnchoredLeft ? containerRect[ANCHOR_RIGHT] : containerRect[ANCHOR_LEFT];
```

Comment:
it's too long, and I preferred having the ternary on one line rather than splitting after the `?`. Which do you think is better?